sorc/ocean_merge.fd/merge_lake_ocnmsk.f90

```diff
@@ -0,0 +1,181 @@
+      slmsk(i,j) = ceiling(land_frac(i,j)) ! ceiling is used for orog smoothing
+    end do
+    end do
+
+    write(flnm,'(4a,i1,a)') trim(atmres),'.',trim(ocnres),'.tile',tile,'.nc'
+    print *,'output=',trim(flnm)
+    call handle_err (nf90_create (path=trim(pth3)//trim(flnm), &
+    cmode=or(NF90_CLOBBER, NF90_64BIT_OFFSET), ncid=ncid4))   ! netcdf3
+
+    call handle_err (nf90_def_dim (ncid4,'lon', lon, dims(1)))
+    call handle_err (nf90_def_dim (ncid4,'lat', lat, dims(2)))
+    call handle_err (nf90_def_var (ncid4,'land_frac', nf90_float, dims(1:2), v1id))
+    call handle_err (nf90_def_var (ncid4,'lake_frac', nf90_float, dims(1:2), v2id))
+    call handle_err (nf90_def_var (ncid4,'lake_depth',nf90_float, dims(1:2), v3id))
+    call handle_err (nf90_def_var (ncid4,'slmsk',     nf90_float, dims(1:2), v4id))
+
+    call handle_err (nf90_enddef  (ncid4))
+
+    call handle_err (nf90_put_var (ncid4, v1id,land_frac))
+    call handle_err (nf90_put_var (ncid4, v2id,lake_frac))
+    call handle_err (nf90_put_var (ncid4, v3id,lake_depth))
+    call handle_err (nf90_put_var (ncid4, v4id,slmsk))
+    call handle_err (nf90_close(ncid4))
+
+!   do i=1,48
+!   do j=1,48
+!     write(*,'(2i4,4f6.1)') i,j,land_frac(i,j),lake_frac(i,j),lake_depth(i,j),slmsk(i,j)
+!   end do
+!   end do
+
+  end do ! tile
+  write(*,'(a,i8,a,i8,a)') 'total lake point ',lake_pt,' where ',nodp_pt,' has no depth'
+
+end program merge_lake_ocnmsk
+
+!> Handle netCDF errors.
+!!
+!! @param[in] ret NetCDF return code.
+!! @author Shan Sun
+subroutine handle_err (ret)
+  use netcdf
+  integer, intent(in) :: ret
+
+  if (ret /= NF90_NOERR) then
+    write(6,*) nf90_strerror (ret)
+    stop 999
+  end if
+end subroutine handle_err
+
+!> Read program namelist.
+!!
+!! @param[out] ocean_mask_dir Directory containing MOM6 ocean mask file.
+!! @param[out] lake_mask_dir Directory containing the lake mask file.
+!! @param[out] out_dir Directory where output file will be written.
+!! @param[out] atmres Atmosphere grid resolution.
+!! @param[out] ocnres Ocean grid resolution.
+!! @author Rahul Mahajan
+!! @author Sanath Kumar
+subroutine read_nml(ocean_mask_dir, lake_mask_dir, atmres, ocnres,out_dir)
+
+  integer :: unit=7, io_status
+
+  character(len=120), intent(out) :: ocean_mask_dir
+  character(len=120), intent(out) :: lake_mask_dir
+  character(len=120), intent(out) :: out_dir
+  character(len=10),  intent(out) :: atmres,ocnres
+
+  !ocean_mask_dir='/scratch1/NCEPDEV/stmp4/Sanath.Kumar/CPLD_GRIDGEN/100/' ! MOM6 mask dir
+  !lake_mask_dir='/scratch1/NCEPDEV/stmp4/Sanath.Kumar/CPLD_GRIDGEN/100/' ! this is for P8
+  !atmres='C96'
+  !ocnres='mx100'
+  !out_dir='/scratch1/BMC/gsd-fv3-dev/fv3data/edit_landfrac_mom6_nov2021/' 
+
+  namelist/mask_nml/ocean_mask_dir, lake_mask_dir, atmres, ocnres, out_dir
+  open(unit=unit, file='input.nml', iostat=io_status )
+  read(unit,mask_nml, iostat=io_status )
+  close(unit)
+  if (io_status > 0) call handle_err(-1, 'Error reading input.nml')
```

Comment:
I am getting a warning here. Routine handle_err only takes one argument. The character sting should be removed from the argument list. That string could be printed before the call to handle_err.

Reply:
removed that and added a print statement to o/p the error message inside the if statement

Reply:
Upon closer inspection, routine handle_err is for netcdf errors. In this case, the minus 1 return code will be converted to a netcdf error message, which will be misleading to the user. Maybe, you should not call that routine and do something like this instead:

```
if (io_status /= 0) then
  print*,"FATAL ERROR reading namelist input.nml"
  stop 999
endif
```

Reply:
Thanks for the insightful tip. Noted. I will make suggested change and test it.  I will also clean up that code in the next commit